fdm-data/src/fertilizers/catalogues/srm.ts

```diff
@@ -1,12 +1,83 @@
-import type { fdmSchema } from "@svenvw/fdm-core"
+import type { CatalogueFertilizer, CatalogueFertilizerItem } from "../d"
 import srm from "./srm.json"
+import xxhash from "xxhash-wasm"
 
-export function getCatalogueSrm(): fdmSchema.fertilizersCatalogueTypeInsert[] {
+const { h32ToString } = await xxhash()
+
+/**
+ * Retrieves the SRM (Sluiting Regionale Kringlopen) fertilizer catalogue.
+ *
+ * This function parses the `srm.json` file and transforms its data into a
+ * structured `CatalogueFertilizer` array. It handles optional properties by
+ * setting them to `null` if they are undefined in the JSON data.
+ *
+ * @returns An array of fertilizer catalogue entries conforming to the
+ *          `CatalogueFertilizer` type.
+ */
+export function getCatalogueSrm(): CatalogueFertilizer {
     const catalogueSrm = srm.map((fertilizer) => {
-        return {
+        const item: CatalogueFertilizerItem = {
             p_source: "srm",
-            ...fertilizer,
+            p_id_catalogue: fertilizer.p_id_catalogue,
+            p_name_nl: fertilizer.p_name_nl,
+            p_name_en: null,
+            p_description: null,
+            p_dm: fertilizer.p_dm === undefined ? null : fertilizer.p_dm,
+            p_density:
+                fertilizer.p_density === undefined
+                    ? null
+                    : fertilizer.p_density,
+            p_om: fertilizer.p_om === undefined ? null : fertilizer.p_om,
+            p_a: fertilizer.p_a === undefined ? null : fertilizer.p_a,
+            p_hc: fertilizer.p_hc === undefined ? null : fertilizer.p_hc,
+            p_eom: fertilizer.p_eom === undefined ? null : fertilizer.p_eom,
+            p_eoc: fertilizer.p_eoc === undefined ? null : fertilizer.p_eoc,
+            p_c_rt: null,
+            p_c_of: fertilizer.p_c_of === undefined ? null : fertilizer.p_c_of,
+            p_c_if: null,
+            p_c_fr: fertilizer.p_c_fr === undefined ? null : fertilizer.p_c_fr,
+            p_cn_of:
+                fertilizer.p_cn_of === undefined ? null : fertilizer.p_cn_of,
+            p_n_rt: fertilizer.p_n_rt === undefined ? null : fertilizer.p_n_rt,
+            p_n_if: fertilizer.p_n_if === undefined ? null : fertilizer.p_n_if,
+            p_n_of: fertilizer.p_n_of === undefined ? null : fertilizer.p_n_of,
+            p_n_wc: fertilizer.p_n_wc === undefined ? null : fertilizer.p_n_wc,
+            p_p_rt: fertilizer.p_p_rt === undefined ? null : fertilizer.p_p_rt,
+            p_k_rt: fertilizer.p_k_rt === undefined ? null : fertilizer.p_k_rt,
```

Comment:
these are pandex names? i remember somewhere else that sometimes p2o5 is used but here it is p_p_rt. might be a pandex issue. for now fine. but when using labels it is the oxide form i expect.

Reply:
Yes, pandex definition is used here. So it is g P2O5 / kg